pkg/splunk/enterprise/indexercluster.go

```diff
@@ -136,6 +136,22 @@ func ApplyIndexerCluster(client splcommon.ControllerClient, cr *enterpriseApi.In
 
 	// no need to requeue if everything is ready
 	if cr.Status.Phase == splcommon.PhaseReady {
+		//update MC
+		//Retrieve monitoring  console ref from CM Spec
+		cmMonitoringConsoleConfigRef, err := RetrieveCMSpec(client, cr, cr.Spec.ClusterMasterRef.Name)
+		if cmMonitoringConsoleConfigRef != "" {
+			namespacedName := types.NamespacedName{Namespace: cr.GetNamespace(), Name: GetSplunkStatefulsetName(SplunkMonitoringConsole, cmMonitoringConsoleConfigRef)}
+			_, err := splctrl.GetStatefulSetByName(client, namespacedName)
+			//if MC pod already exists
+			if err == nil {
+				c := mgr.getMonitoringConsoleClient(cr, cmMonitoringConsoleConfigRef)
+				err := c.AutomateMCApplyChanges(false)
+				return result, err
+			}
+			if cr.Spec.MonitoringConsoleRef.Name != "" || cr.Spec.MonitoringConsoleRef.Name != cmMonitoringConsoleConfigRef {
```

Comment:
Do we really allow mcRef to be configured on Indexer cluster CR? Even if it matches one in CM, I am just worried it will cause confusion.. any thoughts?

Reply:
This "monitoringConsoleRef" is a part of commonSplunkSpec it is available for indexer CR also but what we are doing here is we always check to which MC is CM connected to and attach Indexer with it but even if the customer provides monitoringConsoleRef in indexer CR that will be ignored. I believe to avoid this I will update the MC documentation for indexer CR. Same conclusion we had last time https://github.com/splunk/splunk-operator/pull/403#discussion_r667141568